src/test/cpp/test/FinalRelease.cpp

```diff
@@ -0,0 +1,58 @@
+#include "pch.h"
+#include "winrt/Windows.Foundation.h"
+
+using namespace winrt;
+using namespace Windows::Foundation;
+
+namespace
+{
+    struct Sample : implements<Sample, IStringable>
+    {
+        hstring ToString()
+        {
+            return L"Sample";
+        }
+
+        ~Sample()
+        {
+            // It's safe to QI/AddRef/Release inside destructor.
+            IStringable s;
+            check_hresult(QueryInterface(guid_of<IStringable>(), put_abi(s)));
```

Comment:
Some test cases involving IWeakReferenceSource and IWeakReference would be more valuable here.

Reply:
Well IStringable proves that QI et al works, but agreed that adding test for IWeakReferenceSource is important to show that Resolve does *not* work. 😉